x-pack/plugins/security/server/lib/authentication/authenticator.js

```diff
@@ -234,19 +234,22 @@ class Authenticator {
   /**
    * Returns provider iterator where providers are sorted in the order of priority (based on the session ownership).
    * @param {Object} sessionValue Current session value.
+   * @param {Hapi.Request} request HapiJS request instance.
    * @returns {Iterator.<Object>}
    */
-  *_providerIterator(sessionValue) {
-    // If there is no session to predict which provider to use first, let's use the order
-    // providers are configured in. Otherwise return provider that owns session first, and only then the rest
-    // of providers.
-    if (!sessionValue) {
+  *_providerIterator(sessionValue, request) {
+    // If there is no way to predict which provider to use first, let's use the order providers are configured in.
+    // Otherwise return provider that either owns session or can handle 3rd-party login request first, and only then
+    // the rest of providers.
+    const shouldHandleSAMLResponse = this._isSAMLResponse(request) && this._providers.has('saml');
```

Comment:
__note:__ definitely not the best approach, but we don't have `login` method support in this Kibana version yet. Generalizing `request.loginAttempt` (used only for username/password right now) feels a bit more complex/risky to me.

Reply:
Good, it turns out we're sharing the same concern, I'll handle it with the additional check for `/api/security/v1/saml` as you suggested below.